cmd/kyma/alpha/create/module/module.go

```diff
@@ -333,85 +331,83 @@ func (cmd *command) Run(ctx context.Context) error {
 		}
 	}
 
-	cmd.NewStep("Creating module archive")
-
+	cmd.NewStep("Creating module archive...")
 	archive, err := module.CreateArchive(archiveFS, cmd.opts.ModuleArchivePath, componentDescriptor)
-
+	if err != nil {
+		cmd.CurrentStep.Failure()
+		return err
+	}
 	cmd.CurrentStep.Successf("Module archive created")
 
 	cmd.NewStep("Adding layers to archive...")
-
-	if err := module.AddResources(archive, modDef, l, osFS, cmd.opts.RegistryCredSelector); err != nil {
+	if err = module.AddResources(archive, modDef, l, osFS, cmd.opts.RegistryCredSelector); err != nil {
 		cmd.CurrentStep.Failure()
 		return err
 	}
+	cmd.CurrentStep.Successf("Layers successfully added to archive")
 
-	cmd.CurrentStep.Success()
+	if cmd.opts.RegistryURL == "" {
+		return nil
```

Comment:
`return nil` is not expected, right?

Reply:
It is, the flow just got inverted since this is the last step it can be skipped if there is no reg given

Reply:
but if no regstry given, this create module command is meaningless, right? It can't push image to remote.

Reply:
i guess it's like this because the command could also be used for just persisting the archive locally to inspect layers? i don't know, the registry flag is also not mandatory. i don't know the intentions/requirements when this command was implemented, but agree that this should have a second look at some point.
and it's not only the registry flow, also the generation of the template could be interesting to be used without registry, but at the moment the flow ends here.